code/modules/admin/verbs/adminweather.dm

```diff
@@ -0,0 +1,78 @@
+ADMIN_VERB(run_weather, R_ADMIN|R_FUN, "Run Weather", "Triggers specific weather on the z-level you choose.", ADMIN_CATEGORY_WEATHER)
+
+	var/static/list/weather_choices = list()
+	if(!length(weather_choices))
+		for(var/datum/weather/weather_type as anything in subtypesof(/datum/weather))
+			weather_choices[initial(weather_type.type)] = weather_type
+
+	var/datum/weather/weather_choice = tgui_input_list(user, "Choose a weather to run", "Weather", weather_choices)
+	if(!weather_choice)
+		return
+	weather_choice = weather_choices[weather_choice]
+
+	var/turf/current_turf = get_turf(user.mob)
+	var/z_level = tgui_input_number(user, "Z-Level to target", "Z-Level", min_value = 1, max_value = world.maxz, default = current_turf?.z)
+	if(!isnum(z_level))
+		return
+
+	var/static/list/custom_options = list("Default", "Custom", "Cancel")
+	var/custom_choice = tgui_alert(user, "How would you like to run the weather settings?", "Custom Weather", custom_options)
+	switch(custom_choice)
+		if("Default")
+			SSweather.run_weather(weather_choice, z_level) // default settings
+			return
+		if("Cancel")
+			return
+		//if("Custom") venture on lil fella
+
+	var/static/list/area_choices = list()
+	if(!length(area_choices))
+		for(var/area/area_type as anything in typesof(/area))
+			area_choices[initial(area_type.type)] = area_type
+
+	var/area/area_choice = tgui_input_list(user, "Select an area for weather to target", "Target Area", area_choices)
+	if(!area_choice)
+		return
+	area_choice = area_choices[area_choice]
+
+	var/weather_bitflags = input_bitfield(
+		user,
+		"Weather flags - Select the flags for your weather event",
+		"weather_flags",
+		weather_choice::weather_flags,
+	)
+
```

Comment:
shouldn't these come from the type?

Reply:
Yes, it's whichever weather type the admin selects. This allows you to edit the bitflags to toggle things like weather affecting mobs, turfs, thunder, etc.